### PR TITLE
feat(connector): [Forte] Use connector_response_reference_id as reference to merchant

### PR DESCRIPTION
**Pull request by AaronTJDev**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Addreses issue #2330 

Updated `connector_response_reference_id` to use `transaction_id` from Forte response.

Forte 'POST' Payment Docs -> https://restdocs.forte.net/#c35427927-d955-4087-88d3-f99413ed91c2

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

This code was tested manually, I was able to verify the updated `connector_response_reference_id` in the response to the POST request to create a payment.

forte_response

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/connector/forte/transformers.rs [hunk 1]
```diff
@@ -252,14 +252,16 @@ impl<F, T>
         Ok(Self {
             status: enums::AttemptStatus::foreign_from((response_code, action)),
             response: Ok(types::PaymentsResponseData::TransactionResponse {
-                resource_id: types::ResponseId::ConnectorTransactionId(transaction_id.to_string()),
+                resource_id: types::ResponseId::ConnectorTransactionId(
+                    transaction_id.clone().to_string(),
```
Comment (srujanchikke) [inline crates/router/src/connector/forte/transformers.rs:256]: In Rust, the `to_string()` method is used to convert a value into a String type. When you call `to_string()` on a value, it creates a new String instance and copies the content of the value into that new String. This means that you get a new String that is a clone of the original value.
So clone is not required here

File: crates/router/src/connector/forte/transformers.rs [hunk 2]
```diff
@@ -252,14 +252,16 @@ impl<F, T>
...
+                ),
                 redirection_data: None,
                 mandate_reference: None,
                 connector_metadata: Some(serde_json::json!(ForteMeta {
                     auth_id: item.response.authorization_code,
                 })),
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: Some(transaction_id.clone().to_string()),
```
Comment (srujanchikke) [inline crates/router/src/connector/forte/transformers.rs:264]: Same here and Please change it in other places as well .

Comment (AaronTJDev) [reply crates/router/src/connector/forte/transformers.rs:264]: Thanks for the feedback @srujanchikke , that's good to know!

Comment (srujanchikke) [reply crates/router/src/connector/forte/transformers.rs]: Hi @AaronTJDev 
Please use following cargo commands - 
```
cargo check --all-features                                                                  
cargo clippy --all-features
```
ref - https://github.com/juspay/hyperswitch/blob/main/docs/CONTRIBUTING.md#cargo-commands, and fix the errors thrown by clippy for the checks to pass

Please run `cargo +nightly fmt` to fix failing formatting check

**Review by srujanchikke — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]
Looks good to me!

Thanks for the PR, @AaronTJDev!

**Comment by VedantKhairnar** [comment]
Hey @AaronTJDev  👋,
Thanks a bunch for all your contributions! We've got a little something for you to show our appreciation.
Just take a moment to fill out this [form](https://forms.gle/yqbtjFuP6oxLaBKN8), and get ready for some awesome swag coming your way.
Thanks!

**Comment by AaronTJDev** [comment]
Oh wow, thank you, I appreciate it! @VedantKhairnar

**Pull request merged**